x-pack/plugins/triggers_actions_ui/public/application/components/builtin_action_types/swimlane/steps/swimlane_fields.tsx

```diff
@@ -0,0 +1,338 @@
+  value: field.id,
+});
+
+const createSelectedOption = (field: SwimlaneFieldMappingConfig | null | undefined) =>
+  field != null ? [formatOption(field)] : EMPTY_COMBO_BOX_ARRAY;
+
+interface Props {
+  action: SwimlaneActionConnector;
+  editActionConfig: (property: string, value: any) => void;
+  updateCurrentStep: (step: number) => void;
+  fields: SwimlaneFieldMappingConfig[];
+  errors: IErrorObject;
+}
+
+const connectorTypeButtons = [
+  { id: 'all', label: 'All' },
+  { id: 'alerts', label: 'Alerts' },
+  { id: 'cases', label: 'Cases' },
+];
+
+const SwimlaneFieldsComponent: React.FC<Props> = ({
+  action,
+  editActionConfig,
+  updateCurrentStep,
+  fields,
+  errors,
+}) => {
+  const { mappings, connectorType = SwimlaneConnectorType.All } = action.config;
+  const prevConnectorType = useRef<SwimlaneConnectorType>(connectorType);
+  const hasChangedConnectorType = connectorType !== prevConnectorType.current;
+
+  const [fieldTypeMap, fieldIdMap] = useMemo(
+    () =>
+      fields.reduce(
+        ([typeMap, idMap], field) => {
+          if (field != null) {
+            typeMap.set(field.fieldType, [
+              ...(typeMap.get(field.fieldType) ?? []),
+              formatOption(field),
+            ]);
+            idMap.set(field.id, field);
+          }
+
+          return [typeMap, idMap];
+        },
+        [
+          new Map<string, Array<EuiComboBoxOptionOption<string>>>(),
+          new Map<string, SwimlaneFieldMappingConfig>(),
+        ]
+      ),
+    [fields]
+  );
+
+  const textOptions = useMemo(() => fieldTypeMap.get('text') ?? [], [fieldTypeMap]);
+  const commentsOptions = useMemo(() => fieldTypeMap.get('comments') ?? [], [fieldTypeMap]);
+
+  const state = useMemo(
+    () => ({
+      alertSourceConfig: createSelectedOption(mappings?.alertSourceConfig),
+      alertIdConfig: createSelectedOption(mappings?.alertIdConfig),
+      severityConfig: createSelectedOption(mappings?.severityConfig),
+      ruleNameConfig: createSelectedOption(mappings?.ruleNameConfig),
+      caseIdConfig: createSelectedOption(mappings?.caseIdConfig),
+      caseNameConfig: createSelectedOption(mappings?.caseNameConfig),
+      commentsConfig: createSelectedOption(mappings?.commentsConfig),
+      descriptionConfig: createSelectedOption(mappings?.descriptionConfig),
+    }),
+    [mappings]
+  );
+
+  const mappingErrors: Record<string, string> = useMemo(
+    () => (Array.isArray(errors?.mappings) ? errors?.mappings[0] : {}),
+    [errors]
+  );
+
+  const resetConnection = useCallback(() => {
+    updateCurrentStep(1);
+  }, [updateCurrentStep]);
+
+  const editMappings = useCallback(
+    (key: keyof SwimlaneMappingConfig, e: Array<EuiComboBoxOptionOption<string>>) => {
+      if (e.length === 0) {
+        const newProps = {
+          ...mappings,
+          [key]: null,
+        };
+        editActionConfig('mappings', newProps);
+        return;
+      }
+
+      const option = e[0];
```

Comment:
Are we doing `e[0]` here because multiple fields could have the same name?

Reply:
Yes, that's what the `onChange` of `EuiComboBox` provides.